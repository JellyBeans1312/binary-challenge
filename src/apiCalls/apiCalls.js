export const getEvents = async () => {
  try {
    const options = { 
      method: 'GET',
      headers: {
        Authorization: 'Bearer qJhnqW5Rsc36m6cUYXZyCaV2BaFNB0QoGpr4fmk_',
        Accept: 'application/json',
        'Content-Type': 'application/json'
      }
    }

    const response = await fetch('https://api.predicthq.com/v1/events/', options)
    const result = await response.json()
    return result.results
  } catch(error) {
    console.log(error);
  }
}

export const searchEvent = async (searchParam) => {
  try {
    const options = {
      method: 'GET',
      headers: {
        Authorization: 'Bearer qJhnqW5Rsc36m6cUYXZyCaV2BaFNB0QoGpr4fmk_',
        Accept: 'application/json',
        'Content-Type': 'application/json'
      }
    }
    
    const response = await fetch(`https://api.predicthq.com/v1/events/?q=${searchParam}/`, options)
    const result = await response.json()
    return result.results

  } catch(error) {
    throw new Error(error)
  }
}

export const getAddress = async (lat, long) => {
  try {
    const response = await fetch(`https://maps.googleapis.com/maps/api/geocode/json?latlng=${lat},${long}&key=AIzaSyDPPlNyjATG8sQnfsdg45ln7kUoyFH24m0`)
    const result = await response.json();
    return result.results.map(result => {
      return result.formatted_address
    });
  } catch(error) {
    console.log(error)
  }
}