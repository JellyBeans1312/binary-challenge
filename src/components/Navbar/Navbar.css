.nav {
  height: 10vh;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  justify-items: center;
}

.nav-button {
  font-size: 2rem;
  border: 2px solid black;
  border-radius: 10px;
  width: 70%;
  margin: auto 0;
}