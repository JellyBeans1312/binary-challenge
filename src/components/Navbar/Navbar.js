import React from 'react';
import { NavLink, Route } from 'react-router-dom'
import  FilterEvents   from '../../containers/FilterEvents/FilterEvents'
import  EventContainer  from '../EventContainer/EventContainer';
import  SearchLocation  from '../../containers/SearchLocation/SearchLocation';

export const Navbar = () => {
  return (
    <>
      <nav>
      <NavLink exact to='/'>All Events</NavLink>
      <NavLink exact to='/near-you'>Search events by location</NavLink>
      <NavLink exact to='filter-events'>Filter</NavLink>
      </nav>
      <section>
        <Route exact path='/' component={EventContainer}/>
        <Route exact path='/near-you' component={SearchLocation}/>
        <Route exact path='/filter-events' component={FilterEvents}/>
      </section>
    </>

  )
}

