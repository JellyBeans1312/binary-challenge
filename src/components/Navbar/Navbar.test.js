import React from 'react';
import { shallow } from 'enzyme';
import { Navbar } from './Navbar';
import { getEvents } from '../../apiCalls/apiCalls';

jest.mock('../../apiCalls/apiCalls')

describe('Navbar', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Navbar addEvents={jest.fn()}/>)
    expect(wrapper).toMatchSnapshot()
  });

  it('should call getEvents onClick', () => {
    const wrapper = shallow(<Navbar addEvents={jest.fn()}/>)
    wrapper.find('NavLink').at(0).simulate('click', {preventDefault: jest.fn()})
    const mockEvents = [
      { title: "something", category: 'concert', location: [{lat: 23, long: 234}] }
    ];
    getEvents.mockImplementation(() => {
      return Promise.resolve({mockEvents})
    });
    expect(getEvents).toHaveBeenCalled()    
  });
});