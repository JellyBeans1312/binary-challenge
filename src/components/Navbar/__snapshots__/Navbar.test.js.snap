// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should match snapshot 1`] = `
<Fragment>
  <nav
    className="nav"
  >
    <NavLink
      className="nav-button"
      exact={true}
      onClick={[Function]}
      to="/"
    >
      All Events
    </NavLink>
    <NavLink
      className="nav-button"
      exact={true}
      to="/near-you"
    >
      Search events by location
    </NavLink>
    <NavLink
      className="nav-button"
      exact={true}
      to="filter-events"
    >
      Filter
    </NavLink>
  </nav>
  <section>
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(EventContainer)",
          "type": [Function],
        }
      }
      exact={true}
      path="/"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(SearchLocation)",
          "type": [Function],
        }
      }
      exact={true}
      path="/near-you"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(FilterEvents)",
          "type": [Function],
        }
      }
      exact={true}
      path="/filter-events"
    />
  </section>
</Fragment>
`;
